src/views/ApartmentApp/components/StudentApplication/components/ApplicantList/components/ApplicantListItem/index.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Grid,
+  Divider,
+  ListItem,
+  ListItemAvatar,
+  ListItemSecondaryAction,
+  ListItemText,
+  Avatar,
+  IconButton,
+} from '@material-ui/core';
+import ClearIcon from '@material-ui/icons/Clear';
+import PersonIcon from '@material-ui/icons/Person';
+import StarBorderIcon from '@material-ui/icons/StarBorder';
+import StarIcon from '@material-ui/icons/Star';
+import user from '../../../../../../../../services/user';
+
+/**
+ * @typedef { import('../../../../services/user').StudentProfileInfo } StudentProfileInfo
+ */
+
+// Based off src/views/PeopleSearch/components/PeopleSearchResult
+// but using props.profile of type StudentProfileInfo
+// rather than using Person of type PeopleSearchResult
+const ApplicantListItem = ({
+  disabled,
+  profile,
+  isApplicationEditor,
+  onChangeEditor,
+  onApplicantRemove,
+}) => {
+  const [avatar, setAvatar] = useState(null);
+  const [hasNickName, setHasNickname] = useState(false);
+  const [personClass, setPersonClass] = useState(profile.Class);
+
+  useEffect(() => {
+    loadAvatar(profile);
+    createNickname(profile);
+    if (String(profile.PersonType).includes('stu') && profile.Class !== undefined) {
+      setPersonClass(profile.Class);
+    } else {
+      // Technically, this case should never happen because the list does not allow the user to add a non-student to the applicant list
+      setPersonClass('');
+    }
+  }, [profile]);
+
+  /**
+   * Creates the Avatar image of the given user
+   * @param {StudentProfileInfo} profile The StudentProfileInfo object for the student represented by this list item
+   */
+  const loadAvatar = async (profile) => {
+    try {
+      const { def: defaultImage, pref: preferredImage } = await user.getImage(profile.AD_Username);
+      setAvatar(preferredImage || defaultImage);
+    } catch {
+      setAvatar(
+        <svg width="50" height="50" viewBox="0 0 50 50">
+          <rect width="50" height="50" rx="10" ry="10" fill="#CCC" />
+        </svg>,
+      );
+    }
+  };
+
+  /**
+   * Saves the nickname of the given user if available
+   * @param {StudentProfileInfo} profile The StudentProfileInfo object for the student represented by this list item
+   */
+  const createNickname = (profile) => {
+    let Name = String(profile.fullName);
+    let FirstName = Name.split(' ')[0];
+    setHasNickname(FirstName !== profile.NickName && profile.NickName !== '');
```

Comment:
`profile.FirstName` is what you want here I think. `profile.fullName` is a frontend-added property that is simply composed of Profile.FirstName + ' ' + Profile.LastName

Reply:
I think once you have the simplified version, this helper function is no longer necessary, since it's only one line, and is only used in the useEffect above.
```suggestion
    setHasNickname(profile.FirstName !== profile.NickName && profile.NickName !== '');
```

Reply:
I think I wrote that (or, actually, copy and paste from somewhere else within 360) that code last semester, and I didn't know that profile.FirstName existed at the time

So yes, I agree this whole nickname handling section needs some cleaning as well